include/nntile/base_types.hh

```diff
@@ -15,49 +15,376 @@
 #pragma once
 
 #include <cstdint>
+#include <assert.h>
+#include <iostream>
 
 namespace nntile
 {
 
-//! Large enough signed integer for indexing purposes
-using Index = int64_t;
+//! Enumeration of base supported data types inside tiles and tensors
+//enum class DTypeEnum: int
+//{
+//    NOT_INITIALIZED=101,
+//    INT64,
+//    INT32,
+//    INT16,
+//    INT8,
+//    BOOL,
+//    FP64,
+//    FP32,
+//    FP32_FAST_TF32,
+//    FP32_FAST_FP16,
+//    FP32_FAST_BF16,
+//    TF32,
+//    FP16,
+//    BF16,
+//    FP8_E4M3,
+//    FP8_E5M2
+//};
+//    constexpr char *name()
+//    {
+//        switch(value)
+//        {
+//            case INT64:
+//                return "int64";
+//                break;
+//            case INT32:
+//                return "int32";
+//                break;
+//            case INT16:
+//                return "int16";
+//                break;
+//            case INT8:
+//                return "int8";
+//                break;
+//            case BOOL:
+//                return "bool";
+//                break;
+//            case FP64:
+//                return "fp64";
+//                break;
+//            case FP32:
+//                return "fp32";
+//                break;
+//            case FP32_FAST_TF32:
+//                return "fp32_fast_tf32";
+//                break;
+//            case FP32_FAST_FP16:
+//                return "fp32_fast_fp16";
+//                break;
+//            case FP32_FAST_BF16:
+//                return "fp32_fast_bf16";
+//                break;
+//            case TF32:
+//                return "tf32";
+//                break;
+//            case FP16:
+//                return "fp16";
+//                break;
+//            case BF16:
```

Comment:
The current changes revise the simple types usage inside kernels and simplify read/write from numpy operations. However, preparation of more tricky types, where internal_t and compat_t types are not the same are less clear

Reply:
I have added a tricky BrainFloat16 type in the last commit (nntile::bf16_t) in files `include/nntile/base_types.hh` and `src/base_types.cc`.